File: engine/sdk/py-kit-engine/kit_engine/peers.py
"""Peer discovery client."""

from __future__ import annotations

import requests


class PeerClient:
    """REST client for kit peer endpoints."""

    def __init__(self, base_url: str, token: str | None = None):
        self._url = f"{base_url}/peers"
        self._token = token

    def _headers(self) -> dict[str, str]:
        if not self._token:
            return {}
        return {"Authorization": f"Bearer {self._token}"}

    def list(self) -> list[dict]:
        resp = requests.get(self._url)
        resp.raise_for_status()
        return resp.json()

    def trust(self, id: str) -> None:
        resp = requests.post(f"{self._url}/{id}/trust", headers=self._headers())
        resp.raise_for_status()

    def block(self, id: str) -> None:
        resp = requests.post(f"{self._url}/{id}/block", headers=self._headers())
        resp.raise_for_status()

File: engine/sdk/ts-kit-engine/src/peers.ts
export interface PeerInfo { id: string; name?: string; address: string; trusted: boolean }

export class PeerClient {
  constructor(private baseURL: string, private token?: string) {}

  async list(): Promise<PeerInfo[]> {
    const res = await fetch(`${this.baseURL}/peers`);
    if (!res.ok) throw new Error(`peers list: ${res.status}`);
    return (await res.json()) as PeerInfo[];
  }

  async trust(id: string): Promise<void> {
    const res = await fetch(`${this.baseURL}/peers/${encodeURIComponent(id)}/trust`, {
      method: "POST",
      headers: this.token ? { Authorization: `Bearer ${this.token}` } : undefined,
    });
    if (!res.ok) throw new Error(`peers trust: ${res.status}`);
  }

  async block(id: string): Promise<void> {
    const res = await fetch(`${this.baseURL}/peers/${encodeURIComponent(id)}/block`, {
      method: "POST",
      headers: this.token ? { Authorization: `Bearer ${this.token}` } : undefined,
    });
    if (!res.ok) throw new Error(`peers block: ${res.status}`);
  }
}
